flake.nix

```diff
@@ -94,157 +92,166 @@
           '';
         };
 
+        # Custom packages and aliases for building hosts
+        # Accessible through 'nix build', 'nix run', etc
         packages = with flake.nixosConfigurations; {
-          "homestakeros" = homestakeros.config.system.build.kexecTree;
+          "homestaker" = homestaker.config.system.build.kexecTree;
         };
       };
       flake =
         let
           inherit (self) outputs;
 
-          homestakeros_options = with nixpkgs.lib; {
-            localization = {
-              hostname = mkOption {
-                type = types.str;
-                default = "homestaker";
-                description = "The name of the machine";
-              };
-              timezone = mkOption {
-                type = types.str;
-                default = "Europe/Helsinki";
-                description = "The time zone used when displaying times and dates";
+          homestakerOptions = with nixpkgs.lib;
+            let
+              nospace = str: filter (c: c == " ") (stringToCharacters str) == [ ];
+            in
+            {
+              localization = {
+                hostname = mkOption {
+                  type = types.strMatching "^$|^[[:alnum:]]([[:alnum:]_-]{0,61}[[:alnum:]])?$";
+                  default = "homestaker";
+                  description = "The name of the machine.";
+                };
+                timezone = mkOption {
+                  type = types.nullOr (types.addCheck types.str nospace);
+                  default = null;
+                  description = "The time zone used when displaying times and dates.";
+                  example = "America/New_York";
+                };
               };
-            };
-
-            mounts = mkOption {
-              type = types.attrsOf types.attrs;
-              default = { };
-              description = "Systemd mounts configuration";
-            };
 
-            ssh = {
-              privateKeyPath = mkOption {
-                type = types.path;
-                default = "/var/mnt/secrets/ssh/id_ed25519";
-                description = "Path, where SSH host keys are automatically generated";
+              mounts = mkOption {
+                type = types.attrsOf types.attrs;
+                default = { };
+                description = "Systemd mounts configuration.";
```

Comment:
This description should also include a documentation link for systemd.mounts https://www.freedesktop.org/software/systemd/man/systemd.mount.html#Options

This is in case we want to later also include the descriptions of each option as a part of the web UI.